lib/systems/default.nix

```diff
@@ -1,9 +1,87 @@
-{ lib }:
-  let inherit (lib.attrsets) mapAttrs; in
+{ lib
+}:
+let
+ inherit (lib.attrsets) mapAttrs;
+
+ selectEmulator = platform: pkgs:
+   let
+     qemu-user = pkgs.qemu.override {
+       smartcardSupport = false;
+       spiceSupport = false;
+       openGLSupport = false;
+       virglSupport = false;
+       vncSupport = false;
+       gtkSupport = false;
+       sdlSupport = false;
+       pulseSupport = false;
+       pipewireSupport = false;
+       smbdSupport = false;
+       seccompSupport = false;
+       enableDocs = false;
+       hostCpuTargets = [ "${platform.qemuArch}-linux-user" ];
+     };
+     wine = (pkgs.winePackagesFor "wine${toString platform.parsed.cpu.bits}").minimal;
+   in
+     if lib.systems.canExecute pkgs.stdenv.hostPlatform platform
+     then "${pkgs.runtimeShell} -c '\"$@\"' --"
+     else if platform.isWindows
+     then "${wine}/bin/wine${lib.optionalString (platform.parsed.cpu.bits == 64) "64"}"
+     else if platform.isLinux && pkgs.stdenv.hostPlatform.isLinux && platform.qemuArch != null
+     then "${qemu-user}/bin/qemu-${platform.qemuArch}"
+     else if platform.isWasi
+     then "${pkgs.wasmtime}/bin/wasmtime"
+     else if platform.isMmix
+     then "${pkgs.mmixware}/bin/mmix"
+     else null;
 
-rec {
   doubles = import ./doubles.nix { inherit lib; };
+
   parse = import ./parse.nix { inherit lib; };
+
+  # uncomment after 23.11 branch-off
```

Comment:
To be kind to external users supporting both stable and unstable, these shouldn't be made warnings until after 24.05 is branched off, and errors after 24.11 is branched off, right?

Reply:
> To be kind to external users supporting both stable and unstable, these shouldn't be made warnings until after 24.05 is branched off, and errors after 24.11 is branched off, right?

I was hoping to get the `lib.warn` in before the 24.05 branch-off, since `lib.warn` is only a hard CI failure for `nixpkgs` itself.

Since `lib.warn` is our way of giving a heads-up to people downstream of us, anybody who has their CI set up to fail if `nixpkgs` emits a warning is basically turning anything sent across our only communication channel into breakage.  I think that's a bad choice and I don't think many users choose it (for very long).

So ideally I'd like to put the `lib.warn` in *before* 24.05.

The timeline for the removal of these attributes is less urgent; if that needs to wait until *after* 24.11 I guess that's okay.  Keep in mind however that until these attributes are removed, we have no way to `lib.warn` on any deprecated attributes of system objects (like `isEfi`).

Ultimately, if people really want even the `lib.warn` to wait until after 24.05, then I guess it has to wait.

Even though this PR has two approvals I will not merge it myself.

Reply:
Alright, let me know if 1f7f6d396241c54829df4209c383a1aa2062c35a captures this intent correctly.